src/ParallelAlgorithms/Interpolation/Actions/InterpolatorReceivePoints.hpp

```diff
@@ -80,12 +80,32 @@ struct ReceivePoints {
                                          Metavariables>>(*vars_holders)
                   .infos;
 
-          // Add the target interpolation points at this temporal_id.
-          vars_infos.emplace(std::make_pair(
-              temporal_id,
-              intrp::Vars::Info<VolumeDim, typename InterpolationTargetTag::
-                                               vars_to_interpolate_to_target>{
-                  std::move(block_logical_coords)}));
+          // Add the new target interpolation points at this temporal_id. If we
+          // already have some target points at this temporal_id, we overwrite
+          // them with the new target points. This is because the interpolation
+          // has already finished for the previous set of target points if we
+          // are receiving new target points. This core must not have had any
+          // elements that contained any of the previous target points. However,
+          // these elements may now contain some new target points.
+          if (vars_infos.count(temporal_id) == 0) {
+            vars_infos.emplace(std::make_pair(
```

Comment:
`vars_infos.insert_or_assign`?  (Haven't checked exactly what type this is, but most similar classes have it.)

Reply:
Hmm let me think if we need to keep around the other members of the `Info` class when we receive a new set of points. I'm thinking no.